src/controller/auth/signIn.js

```diff
@@ -0,0 +1,34 @@
+require('env2')('.env');
+const { compare } = require('bcrypt');
+const { getUser } = require('../../database/queries/auth');
+const signInSchema = require('../../utils/validation/login.validation');
+const { generateToken } = require('../../utils/helpers/authToken');
+const CustomError = require('../../utils/helpers/customError');
+
+const signIn = (req, res, next) => {
+  signInSchema
+    .validateAsync(req.body)
+    .then((validated) => getUser(validated.email))
+    .then((userData) => {
+      if (userData.rowCount === 0) {
+        throw new CustomError(400, 'Email is invalid');
+      }
+      // eslint-disable-next-line prefer-destructuring
+      req.user = userData.rows[0];
+      return compare(req.body.password, userData.rows[0].password);
+    })
+    .then((isValid) => {
+      if (!isValid) throw new CustomError(400, 'password is incorrect');
```

Comment:
Line 14 and 21:
Consider using a general error message like 'Invalid email or password' instead of specifying which field was incorrect to improve security and prevent sensitive information leakage.